cmd/upgrade/executor/cstor_spc.go

```diff
@@ -66,20 +69,38 @@ func NewUpgradeCStorSPCJob() *cobra.Command {
 	return cmd
 }
 
+func singleCstorSPCUpgrade(cmd *cobra.Command) {
+	util.CheckErr(options.RunPreFlightChecks(cmd), util.Fatal)
+	util.CheckErr(options.InitializeDefaults(cmd), util.Fatal)
+	util.CheckErr(options.RunCStorSPCUpgrade(cmd), util.Fatal)
+}
+
+func bulkCstorSPCUpgrade(cmd *cobra.Command, args []string) {
+	for _, name := range args {
+		options.cstorSPC.spcName = name
+		singleCstorSPCUpgrade(cmd)
+	}
+}
+
 // RunCStorSPCUpgradeChecks will ensure the sanity of the cstor SPC upgrade options
-func (u *UpgradeOptions) RunCStorSPCUpgradeChecks(cmd *cobra.Command) error {
-	if len(strings.TrimSpace(u.cstorSPC.spcName)) == 0 {
-		return errors.Errorf("Cannot execute upgrade job: cstor spc name is missing")
+func (u *UpgradeOptions) RunCStorSPCUpgradeChecks(cmd *cobra.Command, args []string) error {
+	if len(strings.TrimSpace(u.cstorSPC.spcName)) == 0 && len(args) == 0 {
+		return errors.Errorf("Cannot execute upgrade job:" +
+			" neither spc-name flag is set nor spc name list is provided")
 	}
 
 	return nil
 }
 
 // RunCStorSPCUpgrade upgrades the given Jiva Volume.
 func (u *UpgradeOptions) RunCStorSPCUpgrade(cmd *cobra.Command) error {
+	klog.V(4).Infof("Started upgrading %s{%s} from %s to %s",
```

Comment:
lets have similar message for completed upgrade also

Reply:
Done